src/toxcore/stats.rs

```diff
@@ -0,0 +1,112 @@
+/*!
+Statistics of incoming/outgoing packets
+This is used by both Udp codec and Tcp codec.
+*/
+
+#[cfg(target_pointer_width = "64")]
+use std::sync::atomic::*;
+#[cfg(target_pointer_width = "32")]
+use std::sync::Mutex;
+use std::sync::Arc;
+
+#[derive(Clone, Debug, Default)]
+/// Struct for counting packets on x64 CPU.
+#[cfg(target_pointer_width = "64")]
+pub struct Stats {
+    /// Incoming packets count for Udp/Tcp
+    incoming: Arc<AtomicUsize>,
+    /// Outgoing packets count for Udp/Tcp
+    outgoing: Arc<AtomicUsize>,
+}
+
+#[derive(Clone, Debug, Default)]
+/// Struct for counting packets on x86 CPU.
+#[cfg(target_pointer_width = "32")]
```

Comment:
I think `#[cfg(not(target_pointer_width = "64"))]` is better - theoretically there might be architectures with other widths.